Flaky test: `tests/chainer_tests/functions_tests/math_tests/test_inv.py::BatchInvFunctionTest`

https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1866/TEST=CHAINERX_chainer-py3,label=mn1-p100/console

```
14:53:53 =================================== FAILURES ===================================
14:53:53 _ BatchInvFunctionTest_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_0_{dtype=float16, shape=(5, 1, 1)}.test_backward _
14:53:53 
14:53:53 self = <chainer.testing._bundle.BatchInvFunctionTest_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__...udnn_fast_batch_normalization_false__use_ideep_never_param_0_{dtype=float16, shape=(5, 1, 1)} testMethod=test_backward>
14:53:53 backend_config = <BackendConfig use_chainerx=False chainerx_device=None use_cuda=True cuda_device=1 use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
14:53:53 
14:53:53     def test_backward(self, backend_config):
14:53:53         """Tests backward computation."""
14:53:53 >       self.run_test_backward(backend_config)
14:53:53 
14:53:53 backend_config = <BackendConfig use_chainerx=False chainerx_device=None use_cuda=True cuda_device=1 use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
14:53:53 self       = <chainer.testing._bundle.BatchInvFunctionTest_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__...udnn_fast_batch_normalization_false__use_ideep_never_param_0_{dtype=float16, shape=(5, 1, 1)} testMethod=test_backward>
14:53:53 
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:459: 
14:53:53 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:255: in run_test_backward
14:53:53     do_check()
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:244: in do_check
14:53:53     **self.check_backward_options)
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/contextlib.py:99: in __exit__
14:53:53     self.gen.throw(type, value, traceback)
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:36: in raise_if_fail
14:53:53     cls.fail(message, e)
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:27: in fail
14:53:53     utils._raise_from(cls, message, exc)
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/utils/__init__.py:106: in _raise_from
14:53:53     six.raise_from(new_exc.with_traceback(orig_exc.__traceback__), None)
14:53:53 <string>:3: in raise_from
14:53:53     ???
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:34: in raise_if_fail
14:53:53     yield
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:244: in do_check
14:53:53     **self.check_backward_options)
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:900: in check_backward
14:53:53     detect_nondifferentiable, is_immutable_params=False
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:464: in run
14:53:53     self._run()
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:507: in _run
14:53:53     self._compare_gradients(gx_numeric, gx_backward, directions)
14:53:53 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:53:53 
14:53:53 self = <chainer.gradient_check._CheckBackward object at 0x7fcc695c5048>
14:53:53 gx_numeric = array(-0.00123553), gx_backward = array(-0.00021403)
14:53:53 directions = [array([[[-0.59490847]],
14:53:53 
14:53:53        [[ 0.15857277]],
14:53:53 
14:53:53        [[-0.78538824]],
14:53:53 
14:53:53        [[-0.00122899]],
14:53:53 
14:53:53        [[ 0.06404992]]])]
14:53:53 
14:53:53     def _compare_gradients(self, gx_numeric, gx_backward, directions):
14:53:53         atol = self.atol
14:53:53         rtol = self.rtol
14:53:53         # Compare the gradients
14:53:53         try:
14:53:53             testing.assert_allclose(
14:53:53                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
14:53:53         except AssertionError as e:
14:53:53             eps = self.eps
14:53:53             xs = self.xs
14:53:53             gys = self.gys
14:53:53             f = six.StringIO()
14:53:53             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
14:53:53                 eps, atol, rtol))
14:53:53             for i, x in enumerate(xs):
14:53:53                 f.write('inputs[{}]:\n'.format(i))
14:53:53                 f.write('{}\n'.format(x))
14:53:53             for i, gy in enumerate(gys):
14:53:53                 f.write('grad_outputs[{}]:\n'.format(i))
14:53:53                 f.write('{}\n'.format(gy))
14:53:53             for i, d in enumerate(directions):
14:53:53                 f.write('directions[{}]:\n'.format(i))
14:53:53                 f.write('{}\n'.format(d))
14:53:53             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
14:53:53             f.write('gradients (backward): {}\n'.format(gx_backward))
14:53:53             f.write('\n')
14:53:53             f.write('x: numeric gradient, y: backward gradient')
14:53:53             f.write(str(e))
14:53:53 >           raise AssertionError(f.getvalue())
14:53:53 E           chainer.testing.function_link.FunctionTestError: Parameterized test failed.
14:53:53 E           
14:53:53 E           Base test method: BatchInvFunctionTest_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_backward
14:53:53 E           Test parameters:
14:53:53 E             dtype: <class 'numpy.float16'>
14:53:53 E             shape: (5, 1, 1)
14:53:53 E           
14:53:53 E           
14:53:53 E           (caused by)
14:53:53 E           FunctionTestError: backward is not implemented correctly
14:53:53 E           
14:53:53 E           (caused by)
14:53:53 E           AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
14:53:53 E           inputs[0]:
14:53:53 E           [[[1.001 ]]
14:53:53 E           
14:53:53 E            [[0.9956]]
14:53:53 E           
14:53:53 E            [[0.9985]]
14:53:53 E           
14:53:53 E            [[1.007 ]]
14:53:53 E           
14:53:53 E            [[0.99  ]]]
14:53:53 E           grad_outputs[0]:
14:53:53 E           [[[-0.757 ]]
14:53:53 E           
14:53:53 E            [[ 0.3416]]
14:53:53 E           
14:53:53 E            [[ 0.652 ]]
14:53:53 E           
14:53:53 E            [[-0.7266]]
14:53:53 E           
14:53:53 E            [[ 0.1501]]]
14:53:53 E           directions[0]:
14:53:53 E           [[[-0.59490847]]
14:53:53 E           
14:53:53 E            [[ 0.15857277]]
14:53:53 E           
14:53:53 E            [[-0.78538824]]
14:53:53 E           
14:53:53 E            [[-0.00122899]]
14:53:53 E           
14:53:53 E            [[ 0.06404992]]]
14:53:53 E           gradients (numeric):  -0.0012355287651527407
14:53:53 E           gradients (backward): -0.00021402932363815547
14:53:53 E           
14:53:53 E           x: numeric gradient, y: backward gradient
14:53:53 E           Not equal to tolerance rtol=0.001, atol=0.001
14:53:53 E           
14:53:53 E           Mismatch: 100%
14:53:53 E           Max absolute difference: 0.0010215
14:53:53 E           Max relative difference: 4.77270789
14:53:53 E            x: array(-0.001236)
14:53:53 E            y: array(-0.000214)
14:53:53 E           
14:53:53 E           assert_allclose failed: 
14:53:53 E             shape: () ()
14:53:53 E             dtype: float64 float64
14:53:53 E             i: (0,)
14:53:53 E             x[i]: -0.0012355287651527407
14:53:53 E             y[i]: -0.00021402932363815547
14:53:53 E             relative error[i]: 4.772707889511269
14:53:53 E             absolute error[i]: 0.0010214994415145852
14:53:53 E             relative tolerance * |y[i]|: 2.1402932363815547e-07
14:53:53 E             absolute tolerance: 0.001
14:53:53 E             total tolerance: 0.0010002140293236383
14:53:53 E           x: -0.00123553
14:53:53 E           y: -0.00021403
14:53:53 
14:53:53 atol       = 0.001
14:53:53 d          = array([[[-0.59490847]],
14:53:53 
14:53:53        [[ 0.15857277]],
14:53:53 
14:53:53        [[-0.78538824]],
14:53:53 
14:53:53        [[-0.00122899]],
14:53:53 
14:53:53        [[ 0.06404992]]])
14:53:53 directions = [array([[[-0.59490847]],
14:53:53 
14:53:53        [[ 0.15857277]],
14:53:53 
14:53:53        [[-0.78538824]],
14:53:53 
14:53:53        [[-0.00122899]],
14:53:53 
14:53:53        [[ 0.06404992]]])]
14:53:53 eps        = 0.001
14:53:53 f          = <_io.StringIO object at 0x7fcc695bcd38>
14:53:53 gx_backward = array(-0.00021403)
14:53:53 gx_numeric = array(-0.00123553)
14:53:53 gy         = array([[[-0.757 ]],
14:53:53 
14:53:53        [[ 0.3416]],
14:53:53 
14:53:53        [[ 0.652 ]],
14:53:53 
14:53:53        [[-0.7266]],
14:53:53 
14:53:53        [[ 0.1501]]], dtype=float16)
14:53:53 gys        = (array([[[-0.757 ]],
14:53:53 
14:53:53        [[ 0.3416]],
14:53:53 
14:53:53        [[ 0.652 ]],
14:53:53 
14:53:53        [[-0.7266]],
14:53:53 
14:53:53        [[ 0.1501]]], dtype=float16),)
14:53:53 i          = 0
14:53:53 rtol       = 0.001
14:53:53 self       = <chainer.gradient_check._CheckBackward object at 0x7fcc695c5048>
14:53:53 x          = array([[[1.001 ]],
14:53:53 
14:53:53        [[0.9956]],
14:53:53 
14:53:53        [[0.9985]],
14:53:53 
14:53:53        [[1.007 ]],
14:53:53 
14:53:53        [[0.99  ]]], dtype=float16)
14:53:53 xs         = (array([[[1.001 ]],
14:53:53 
14:53:53        [[0.9956]],
14:53:53 
14:53:53        [[0.9985]],
14:53:53 
14:53:53        [[1.007 ]],
14:53:53 
14:53:53        [[0.99  ]]], dtype=float16),)
14:53:53 
14:53:53 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:537: FunctionTestError
```

--- a/tests/chainer_tests/functions_tests/math_tests/test_inv.py
+++ b/tests/chainer_tests/functions_tests/math_tests/test_inv.py
@@ -113,7 +113,7 @@ class BatchInvFunctionTest(testing.FunctionTestCase):
         if self.dtype == numpy.float16:
             self.check_forward_dtype = numpy.float32
             self.check_forward_options.update({'atol': 1e-3, 'rtol': 1e-3})
-            self.check_backward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+            self.check_backward_options.update({'atol': 2e-3, 'rtol': 2e-3})
             self.check_double_backward_options.update({
                 'atol': 5e-3, 'rtol': 5e-3})
         else:
@@ -124,8 +124,8 @@ class BatchInvFunctionTest(testing.FunctionTestCase):
                 'atol': 1e-3, 'rtol': 1e-3})
 
     def generate_inputs(self):
-        x = (numpy.eye(self.shape[-1]) +
-             numpy.random.uniform(-0.01, 0.01, self.shape)).astype(self.dtype)
+        x = numpy.random.uniform(-0.01, 0.01, self.shape).astype(self.dtype)
+        x += numpy.eye(self.shape[-1])
         return x,
 
     def forward(self, inputs, device):

